### PR TITLE
Redirect rules to wire up the 9.0 guide

### PR DESCRIPTION
**Pull request by janhoy**
The purpose of this is to
* Add a rewrite rule for `/guide/solr/X_Y` to the __root area
* Add an "alias" so that `/guide/9_0` gets redirected (not rewritten) to /guide/solr/9_0
* Adapt the wildcard rule `/guide/[a-z].*` which would earlier rewrite `/guide/solr/xxx` to `/guide/8_11/solr/xxx`, make exception for `solr`. After 9.0 release this rule will likely instead be `/guide/solr/foo` pointing to `/guide/solr/latest/foo`??
* Add preliminary `htaccess.ref-guide-antora` to test the `/guide/solr/latest` -> `/guide/solr/9_0` redirect

This can be committed now, before the release, as there will still be nowhere on our page linking to it. This way we can test the entire new guide on the real site before announcing it.

### CODE REVIEW

File: themes/solr/templates/htaccess.ref-guide-antora [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Redirect 302 /guide/solr/latest /guide/solr/9_0
+Redirect 302 /guide/solr/index.html /guide/solr/9_0/index.html
```
Comment (janhoy) [inline themes/solr/templates/htaccess.ref-guide-antora:2]: Question: When I generated the `.htaccess` file by Antora (solr-ref-guide/build/site/solr/.htaccess), it generated a file like this:
```
Redirect 302 /guide/solr/latest /guide/solr/9_0
Redirect 301 /guide/index.html /guide/solr/9_0/index.html
```

The first one looks correct, but I cannot see how the second one is. The path `/guide/index.html` is currently occupied by our own website as a landing page for all guides (we'll still need that to link to pre-9.0 guides). Also, using a 301 permanent redirect for the index.html to a specific version looks fishy? That's why I changed it to 302 and the path in this PR.

Comment (uschindler) [reply themes/solr/templates/htaccess.ref-guide-antora:2]: Thats wrong, yes! How does it come into play? Maybe the script needs some update?

Comment (janhoy) [reply themes/solr/templates/htaccess.ref-guide-antora:2]: Antora has this built-in, I think it is this feature https://docs.antora.org/antora/latest/playbook/urls-redirect-facility/ but have not looked closely..

Comment (janhoy) [reply themes/solr/templates/htaccess.ref-guide-antora:2]: I tried again, and Antora generates the same 301 rule again. No harm, RMs will just need to hand edit the file before committing to solr-site, until this is fixed. I opened https://issues.apache.org/jira/browse/SOLR-16188 for this, so we don't need to handle it here in this PR.

**Review by uschindler — CHANGES_REQUESTED** [review]

File: themes/solr/templates/htaccess.template [hunk 2]
```diff
@@ -82,7 +82,13 @@ RedirectMatch ^/operator/downloads/crds/(v\d+\.\d+\.\d+)(/?.*)$ https://www.apac
 RewriteRule ^(docs|guide)/\d+(?!.+\.\w+$|.+/$).+$          $0/ [R=301,L]
 
 # __root/docs.solr.apache.org/ is a special alias added by INFRA-19439, so we can refer to stuff in other places like SVN
-RewriteRule ^(docs|guide)/\d+.*$  __root/docs.solr.apache.org/$0 [PT]
+RewriteRule ^docs/\d+.*$  __root/docs.solr.apache.org/$0 [PT]
+# Ref-guides up to 8.x
+RewriteRule ^guide/(6|7|8)_\d+.*$  __root/docs.solr.apache.org/$0 [PT]
+# Redirect v9 and higher guides on format /guide/9_0/foo to /guide/solr/9_0/foo
+RedirectMatch Permanent ^/guide/((9|1\d)_\d.*$) /guide/solr/$1
```
[the remark this answers is not in the document]
Comment (janhoy) [reply themes/solr/templates/htaccess.template]: Ok, removing the redirectmatch, and dealying that particular redirect for now.
Going to merge to staging to test the rules (although the guide will not be visible in staging)

**Comment by HoustonPutman** [comment]
Trying this out on staging, this:

https://solr.staged.apache.org/guide/solr

gets re-routed to:

https://solr.staged.apache.org/guide/8_11/solr/

which is not what we want.

https://solr.staged.apache.org/guide/solr/ works fine, so we just need to make an explicit exception for the `https://solr.staged.apache.org/guide/solr` url, since it is assuming that `solr` is the documentation page.

**Comment by HoustonPutman** [comment]
yeah we probably want to leave `latest` in the address bar, especially because we will have redirects in the future if page names change. I wonder if that helps with our google indexing as well.

**Pull request merged**